homeassistant/components/sensor/folder.py

```diff
@@ -0,0 +1,112 @@
+"""
+Sensor for monitoring the contents of a folder.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.folder/
+"""
+from datetime import datetime as dt
+from datetime import timedelta
+import glob
+import logging
+import os
+
+import voluptuous as vol
+
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.template import DATE_STR_FORMAT
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+
+_LOGGER = logging.getLogger(__name__)
+
+
+CONF_FOLDER_PATHS = 'folder'
+CONF_FILTER = 'filter'
+DEFAULT_FILTER = '*'
+
+SCAN_INTERVAL = timedelta(seconds=10)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_FOLDER_PATHS): cv.isdir,
+    vol.Optional(CONF_FILTER, default=DEFAULT_FILTER): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the folder sensor."""
+    folder = Folder(config.get(CONF_FOLDER_PATHS), config.get(CONF_FILTER))
```

Comment:
All paths have to be whitelisted by checking it with `hass.config.is_allowed_path(path)`